app/grandchallenge/challenges/tasks.py

```diff
@@ -116,3 +121,62 @@ def save_phase():
                 )
 
             save_phase()
+
+
+@acks_late_2xlarge_task
```

Comment:
With some optimisation this can definitely be done on the micro short queue. I think it is best to:

- switch to the micro short queue
- add the `@transaction.atomic` decorator
- filter out challenges with zero relevant tasks with `.exclude(num_is_overdue=0, num_is_overdue_soon=0, num_support_is_overdue=0)`
- rather than iterating over all challenges only iterate over the relevant ones with `.filter(pk__in=onboarding_task_info_by_challenge.keys())`, all of the `get(...)` methods can then be removed.

Reply:
Ah! I hadn't ever traced the emailing to the DB's `RawEmail` creation! Yep. Adding atomic makes perfect sense. I do think having it retry when the lock failed makes sense. I've added that.